flink-autoscaler/src/main/java/org/apache/flink/autoscaler/ScalingExecutor.java

```diff
@@ -264,4 +262,21 @@ private static Map<String, String> getVertexParallelismOverrides(
                 });
         return overrides;
     }
+
+    private boolean blockScalingExecution(
+            Context context,
+            Map<JobVertexID, ScalingSummary> scalingSummaries,
+            Configuration conf,
+            Instant now) {
+        var scaleEnabled = conf.get(SCALING_ENABLED);
+        var isExcluded = CalendarUtils.inExcludedPeriods(conf, now);
+        autoScalerEventHandler.handleScalingEvent(
+                context,
+                scalingSummaries,
+                scaleEnabled,
+                isExcluded,
+                conf.get(SCALING_EVENT_INTERVAL));
+
+      return !scaleEnabled || isExcluded;
```

Comment:
I don't completely understand why we had to move this into a method. It could simply be:
```
       var scaleEnabled = conf.get(SCALING_ENABLED) && !CalendarUtils.inExcludedPeriods(conf, now);
        autoScalerEventHandler.handleScalingEvent(
                context,
                scalingSummaries,
                scaleEnabled,
                conf.get(SCALING_EVENT_INTERVAL));
```
Then we don't need any changes to the autoscaler event handler, we could simply log a line here that we are in an excluded period instead of changing code in multiple places.

Reply:
I've considered just print a log but finally chose adjust autoscaler event handler a bit:

1. Print a log will cause many annoying duplicated logs if exludedPeriods set is relatively wide
2. Now that we have the interval based event trigger method, why not use it

@mxm what do you think

Reply:
I am not a big fan of scattering features across many interfaces without a very good reason. It makes the event handler and the whole logic more complex

Reply:
The event handler really has nothing to do with this feature (and many other things that happen in the scaling executor)

Reply:
I was contemplating to comment on this as well, but after looking at the code, there is some benefit to hinting whether scaling is disabled or blocked. By treating blocked scaling as "disabled", we yield less information to the user. I think it will be beneficial to learn WHY the scaling is disabled, i.e. scaling disabled or scaling blocked.

>The event handler really has nothing to do with this feature (and many other things that happen in the scaling executor)

Not sure about that. Event handlers are supposed to process events and in the context of the autoscaler, a blocked scaling decision is a type of event we may present to the user. We are already leaking whether scaling is enabled or not to the event handler, why not also annotate why it is disabled?

Reply:
If scaling is blocked or disabled the event that we send out is a recommendation. Wouldn't it make sense to have uniform events for recommendation regardless of the cause? The user always knows why it's only a recommendation, either because the autoscaler is disabled completely or if in a blocking period

Reply:
I don't feel strongly about including the cause. We can also just send out scaling events only for actual scaling operations. What @flashJd did is actually more in line with the current code which does include meta information like "scalingEnabled" in the event message. If we decide to skip those, for consistency, I would also remove the "scalingEnabled" flag and only ever emit scaling events when scaling is enabled and not blocked.

Reply:
In the current logic we send 2 types of events on a parallelism change:
 1. Scaling Recommendation (when disabled) -> `Recommended parallelism change: {parallelisms}`
 2. Scaling Event (when enabled) -> `Scaling Vertices: {parallelisms}`
 
The PR basically adds a 3rd type of event which looks like a recommendation but it doesn't say recommendation it prints:

`Scaling execution disabled by config: configkey:false {parallelism:map}`

I don't think that this is fully consistent with the current logic. We intended to keep the recommendations, but we changed the code so that it's not a recommendation anymore but looks different.

Reply:
Ah sorry I did not realise that you also changed the event message for the basic recommendations... my bad

Reply:
Sorry, I misunderstood the code, I am fine with this :)

Reply:
Thanks for the review. A small question, what do you think of the frequent log printing, `Stabilizing until ...` and `Metric window not full until ...`, is it worth to put some frequent log to autoscale event handler.

Reply:
I think it is a good idea to send events for those **once** per stabilization / metric collection period. We have an event de-duplication feature to avoid sending them more often.

Reply:
Let's do that in a separate PR.